sips/sip-02X/sip-02X-wallet-interface.md

```diff
@@ -0,0 +1,213 @@
+The RPC payload is also standardized on its own for use in various contexts.
+
+Backward Compatibility for Wallets: Wallets are encouraged to implement the new webbtc request interface. To maintain backward compatibility, wallets may choose to retain the previous system alongside the new interface. This approach ensures uninterrupted service for applications that have not yet migrated to the new standard.
+Detailed Implementation Guidelines: Detailed guidelines and best practices for integrating the webbtc request standard will be provided. This includes technical specifications, sample code, and migration strategies for both web applications and wallet providers.
+Rationale
+
+The adoption of the webbtc request standard is driven by the need for a more robust and simplified connection protocol between web applications and wallets. This standard provides a modernized approach that aligns with current web development practices, offering better performance, security, and ease of use.
+
+# Backwards Compatibility
+
+The implementation of this proposal is not necessarily backward compatible.
+However, wallets implementing the new standard are advised to maintain the previous system to support legacy applications during a transition period.
+Existing applications using the current Auth system should continue to operate, but immediate changes are recommended once this SIP is ratified.
+
+# Implementation
+
+## Notes on Serialization
+
+The WebBTC request standard is inspired by JSON-RPC 2.0 for serialization.
+To adhere to a generic serializability, the following notes are given.
+Enums are serialized as humand-readable strings.
+BigInts are serialized as numbers, strings, or anything else that can be parsed by the JavaScript BigInt constructor.
+Bytes are serialized as hex-encoded strings (without the 0x prefix).
+Predefined formats from previous SIPs are used where applicable.
+Addresses are serialized as Stacks c32-encoded strings.
+Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+## Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered recommendations for the wallet.
+The user/wallet may choose to ignore/override them.
+Optional params are marked with a `?`.
+
+### Method Independent
+
+#### Transfer & Transaction Definitions
+
+`params`
+
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` (anything parseable by the BigInt constructor)
+- `nonce?`: `number | string` (anything parseable by the BigInt constructor)
+- `attachment?`: `string` hex-encoded
+- `anchoreMode?`: `'onChainOnly' | 'offChainOnly' | 'any'`
+- `postConditionMode?`: `'allow' | 'deny'`
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
```

Comment:
I have tried to align with the JSON API from Xverse, but in some cases I believe it's likely better to decouple from Stacks.js representations of data (post-conditions, clarity-values, etc.)
It should easily be possible for Xverse to add handling for a backward compatible interface. cc @m-aboelenein

Reply:
@m-aboelenein do you think that's something Xverse could support -- I think it's important to stay disconnected from the JavaScript representations _(since those might not exist everywhere or could change)_

Reply:
From the "Notes on Serialization" section, `postConditions` would be typed as a `string[]`?

Reply:
Just saw the "where" section below regarding encoding PostCondition as hex-encoded

Reply:
Yeah, still discussing these. Not sure what's best, but we're thinking for Clarity hex or basic JS objects (similar to Stacks.js, but with readable enums) could work. For post-conditions, I'm not sure if there is a clean solution, the JS repr isn't really nice, but could also work similarly to Clarity.